libcore/src/geometry/Line.cpp

```diff
@@ -218,40 +218,13 @@ Point Line::ShortestPoint(const Point & p) const
         return _point2 + t * lambda;
 }
 
-/*
- *  Prüft, ob Punkt p im Liniensegment enthalten ist
- * algorithm from:
- * http://stackoverflow.com/questions/328107/how-can-you-determine-a-point-is-between-two-other-points-on-a-line-segment
- *
- * */
-bool Line::IsInLineSegment(const Point & p) const
-{
-    Point differenceTwoAndOne = _point2 - _point1;
-    Point differencePAndOne   = p - _point1;
-
-    // cross product to check if point i colinear
-    auto crossProduct = differenceTwoAndOne.CrossProduct(differencePAndOne);
-
-    if(std::abs(crossProduct) > 0.0001) {
-        return false;
-    }
-
-    // dotproduct and distSquared to check if point is in segment and not just in line
-    double dotp = differencePAndOne.ScalarProduct(differenceTwoAndOne);
-
-    if(dotp < 0.) {
-        return false;
-    }
-
-    return dotp <= (differenceTwoAndOne).NormSquare();
-}
-
-bool Line::NearlyInLineSegment(const Point & p) const
+bool Line::IsInLineSegment(const Point & p, double tolerance) const
 {
     return fabs((_point1 - p).Norm() + (_point2 - p).Norm() - (_point2 - _point1).Norm()) <
-           J_EPS_DIST; // old version
+           tolerance;
```

Comment:
If my computations are correct, this still depends on the length of the line.
Example: 
Point p: (0.1, 0.1)
Line line1: (0, 0) - (5,0)
fabs((_point1 - p).Norm() + (_point2 - p).Norm() - (_point2 - _point1).Norm()) = 0.042

Line line2: (0, 0) - (1,0)
fabs((_point1 - p).Norm() + (_point2 - p).Norm() - (_point2 - _point1).Norm()) = 0.047